src/ipv4.c

```diff
@@ -487,7 +487,13 @@ static int ipv4_get_route(struct rtentry *route)
 
 		if (((dest & mask) == (rtdest & rtmask & mask))
 		    && (mask >= route_mask(route).s_addr)
-		    && (mask <= rtmask)) {
+		    && (mask <= rtmask)
+		    && ((route_iface(route) == NULL)
+		        || (strcmp(iface, route_iface(route)) == 0)
+		        || ((strlen(route_iface(route))>0)
```

Comment:
Add spaces around binary operator `>` and perhaps remove some parenthesis, changing:
```
		        || ((strlen(route_iface(route))>0)
		            && (route_iface(route)[0] == '!')
		            && (strcmp(iface, &route_iface(route)[1]) != 0))
```
to:
```
		        || (strlen(route_iface(route)) > 0
		            && route_iface(route)[0] == '!'
		            && strcmp(iface, &route_iface(route)[1]) != 0)
```

Reply:
I have implemented this as suggested before merging